<launch>
    
    <arg name="image_topic"       default="/camera/color/image_raw"/>
    <arg name="camera_info"       default="/camera/color/camera_info"/> 
    <arg name="cornerX"          default="6"/>
    <arg name="cornerY"          default="9"/>    
    <arg name="gridLength"        default="0.028"/> 
    
    <node pkg="hand_eye_calibration" type="checkerboard_detection.py" name="checkerboard_detection">
        <param name="image_topic"        value="$(arg image_topic)"/>
        <param name="camera_info"        value="$(arg camera_info)"/>
        <param name="corners_per_row"    value="$(arg cornerX)"/>
        <param name="corners_per_column" value="$(arg cornerY)"/>
        <param name="grid_length"        value="$(arg gridLength)"/>
    </node> 
    
    <node pkg="image_view" type="image_view" name="image_view" respawn="false" output="screen">
        <remap from="image" to="/checkerboard_detections_image"/>
        <param name="autosize" value="true" />
    </node>

</launch>
